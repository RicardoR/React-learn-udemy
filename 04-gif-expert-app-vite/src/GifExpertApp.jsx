import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {
        setCategories(categoryList => ['New Category', ...categoryList])
    };

    return (
        <>
            <h1> GifExpertApp</h1>
            <button onClick={onAddCategory}>Add category</button>
            <AddCategory/>
            <ol>
                {
                    categories.map((category, index) => <li key={index}>{category}</li>)
                }
            </ol>
        </>
    );
};

