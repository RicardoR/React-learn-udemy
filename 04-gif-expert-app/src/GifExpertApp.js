import { useState } from "react";
import { AddCategory } from './components/AddCategory';

const GifExpertApp = () => {

    const _categories = ['One Punch', 'Samurai x', 'Dragon Ball']
    const [categories, setCategories] = useState(_categories);
        
    return (
      <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr />

        <ol>
          {categories.map((category, idx) => (
            <li key={`${category}-${idx}`}>{category}</li>
          ))}
        </ol>
      </>
    );
};

export default GifExpertApp;