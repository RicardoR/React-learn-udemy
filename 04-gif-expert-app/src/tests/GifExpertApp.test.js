import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('<GifExpertApp />', () => {
    let wrapper = shallow(<GifExpertApp />);

    test('should display', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should display a category list', () => {
        const categories = ['Dragon Ball', 'One Piece'];
        wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
