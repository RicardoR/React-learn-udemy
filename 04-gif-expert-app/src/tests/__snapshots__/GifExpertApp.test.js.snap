// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> should display 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp /> should display a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
